ONSdigital/dp-search-data-finder
# Request urls from zebedee

# What has changed
  
  
  
  
  The handler for the reindex-requested events has been updated so that when an event is consumed it will do the following:
  - use the dp-api-clients-go function, named 'GetPublishedIndex', to call the zebedee endpoint GET /publishedindex and retrieve a list of URLs
  - log out the first ten URLs retrieved from zebedee

  The unit test TestReindexRequestedHandler_Handle has been updated to reflect the changes in the handler.

  A healthcheck has been added for the zebedee client.

  The config and README have been tidied and updated to include the zebedee URL.

  # How to review
  
  Check that the code looks correct and that the following tests pass:

  make test

  go test -component

  make lint

  To check the health do as follows:
  - run this command: curl localhost:28000/health

  To see the list of URLs being logged locally do as follows:
  - in dp-compose run ElasticSearch and Kafka as follows: docker-compose up -d
  - run zebedee
  - run the Search API (but first run this command to use new ElasticSearch): export ELASTIC_SEARCH_URL="http://localhost:11200"
  - run the Search Reindex API
  - call the POST /jobs endpoint on the Search Reindex API to create a reindex-requested event:
  curl -s http://localhost:25700/jobs | jq
  - run the dp-search-data-finder code to see the logs showing ten URLs in the console e.g.
{
  "created_at": "2022-03-09T11:14:13.916987Z",
  "data": {
      "first URLs": [
          "/releases/adoption",
          "/releases",
          "/releases/activitysizeandlocation",
          "/economy/grossdomesticproductgdp/timeseries/ihyq",
          "/economy/grossdomesticproductgdp",
          "/economy/regionalaccounts/grossdisposablehouseholdincome/bulletins/regionalgrossdisposablehouseholdincomegdhi/2014-06-04",
          "/economy/regionalaccounts/grossdisposablehouseholdincome/bulletins/regionalgrossdisposablehouseholdincomegdhi/2013-04-24",
          "/economy/regionalaccounts/grossdisposablehouseholdincome/bulletins/regionalgrossdisposablehouseholdincomegdhi/2015-05-27",
          "/economy/regionalaccounts/grossdisposablehouseholdincome",
          "/economy/regionalaccounts"
      ]
  },
"event": "first 10 URLs retrieved",
"namespace": "dp-search-data-finder",
"severity": 3
}

  # Who can review
  
  Anyone but me
